packages/typespec-client-generator-core/src/types.ts

```diff
@@ -1070,17 +1070,19 @@ function updateTypesFromOperation(
     }
   }
   const lroMetaData = getLroMetadata(program, operation);
-  if (lroMetaData) {
+  if (lroMetaData && generateConvenient) {
```

Comment:
Should `generateConvenient` impact only usage or the model detection? In other logics, it only impact usage.